### PR TITLE
Add docs for portlet, view and viewlets

### PR DESCRIPTION
**Pull request by kakshay21**
see #327 #326 #325

### CODE REVIEW

File: docs/templates/addon/view/index.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    mrbob -O collective.todo bobtemplates.plone:addon
+
+then change into the created folder ``collective.todo`` and create your first View:
+
+.. code-block:: shell
+
+    mrbob bobtemplates.plone:view
+
+It will ask if you need Python class and template file, you can have both or atleat one to have a working view. Based on the input it will ask about class name, template name or both. By default it will suggest you to use class name as view name (the part of url) but you can also change it. You can see your newly created view by using the url that you used for view name on ``IFolderish`` interface.
```
Comment (MrTango) [inline docs/templates/addon/view/index.rst]: at least

File: docs/templates/addon/viewlet/index.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    mrbob -O collective.todo bobtemplates.plone:addon
+
+then change into the created folder ``collective.todo`` and create your first Viewlet:
+
+.. code-block:: shell
+
+    mrbob bobtemplates.plone:viewlet
+
+It will ask if you need Python class and template file, you can have both or atleast one to have a working viewlet. Based on the input it will ask about class name, template name or both. By default it will suggest you to use class name as viewlet name but you can also change it. This will create a viewlet registered to ``IAboveContentTitle`` viewlet manager and on ``IDocument`` interface.
```
Comment (MrTango) [inline docs/templates/addon/viewlet/index.rst]: at least

File: docs/templates/addon/portlet/index.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    Answer with a question mark to display help.
+    Values in square brackets at the end of the questions show the default value if there is no answer.
+
+
+
+    RUN: git status --porcelain --ignore-submodules
+    Git state is clean.
+
+    --> Portlet name to diplay for the portlet [Weather]: My Weather Portlet
```
[the remark this answers is not in the document]
Comment (kakshay21) [reply docs/templates/addon/portlet/index.rst]: oh, it's a typo in the question itself. I'll fix this

**Comment by svx** [comment]
@MrTango you know that your tox setup for the docs, is not testing anything, right ? :)

    sphinx-build -b html -d _build/docs/doctrees docs _build/docs/html -W

Is what is currently in the tox config, and here you tell Sphinx, basically ignore all errors and warnings and "just" build html.
With that config, Sphinx will only fail building, if a add-on is missing or your Sphinx `conf.py` is wrong.

This is not *really* testing :)

If you would include "nitpicky" into tox for the docs (http://www.sphinx-doc.org/en/master/usage/configuration.html#confval-nitpicky) you would have a nice starting point for actually testing :)

More info is also in the "builder docs of Sphinx" (http://www.sphinx-doc.org/en/master/man/sphinx-build.html)

I for example usually have something like this in my config:
`@$(SPHINXBUILD) -j auto -n -W -b html $(ALLSPHINXOPTS) $(BUILDDIR)/debug-strict`

**Comment by kakshay21** [comment]
@MrTango it's good to go anyway.

**Comment by MrTango** [comment]
@svx for the tips, but the current test works somehow. We are treading warnings as errors and it works.
But I'll look into that and try to optimize it.
But feel free to just make the change and PR.

**Pull request merged**